### PR TITLE
Migrate opcache check to new SetupCheck API

### PR DESCRIPTION
**Pull request by come-nc**
* See https://github.com/nextcloud/server/issues/41364

## Summary

Migrate opcache check to new SetupCheck API

## TODO

- [x] Decide what to do with `<code>` tags.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/settings/lib/SetupChecks/PhpOpcacheSetup.php [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+			}
+		}
+
+		return $recommendations;
+	}
+
+	public function run(): SetupResult {
+		$recommendations = $this->getOpcacheSetupRecommendations();
+		if (!empty($recommendations)) {
+			return SetupResult::error(
```
Comment (Altahrim) [inline apps/settings/lib/SetupChecks/PhpOpcacheSetup.php:126]: I think recommendations about cache size can be warning only. 
`save_comment=0` seems to be the only real error here, but I would also include disabled opcache.

Comment (come-nc) [reply apps/settings/lib/SetupChecks/PhpOpcacheSetup.php:126]: I just used the same level as before.
Can you comment inline on recommendations which you think should be warning or error?
Then I can adapt this to a two-level check.

**Review by Altahrim — APPROVED** [review]

**Review by artonge — APPROVED** [review]
Anything changed beside moving code and adapting tests?

**Comment by come-nc** [comment]
Failure not related, see https://github.com/nextcloud/server/pull/42633

**Comment by come-nc** [comment]
> Anything changed beside moving code and adapting tests?

Yes, level is error or warninig depending on which recommendation is triggered, saw that with Benjamin.

**Pull request merged**